frontend/packages/knative-plugin/src/topology/layouts/layoutConstraints.ts

```diff
@@ -68,14 +70,26 @@ const alignNodeConnector = (
           offset: nextOffset + link.source.height / 2,
         });
       }
-      // Keep the event sources to the left
-      constraints.push({
-        axis: 'x',
-        left: link.source.index,
-        right: filteredNode.index,
-        gap: serviceDistance + link.source.width / 2 + options.linkDistance,
-        equality: true,
-      });
+
+      if (type === TYPE_EVENT_SINK_LINK) {
+        // Keep the event sink to the left
+        constraints.push({
+          axis: 'x',
+          left: link.target.index,
+          right: link.source.index,
+          gap: serviceDistance + link.source.width / 2 + options.linkDistance,
+          equality: true,
+        });
+      } else {
+        // Keep the event sources to the left
+        constraints.push({
+          axis: 'x',
+          left: link.source.index,
+          right: filteredNode.index,
+          gap: serviceDistance + link.source.width / 2 + options.linkDistance,
+          equality: true,
+        });
+      }
```

Comment:
we can revert this to 

```suggestion
// Keep the event sources to the left
      constraints.push({
        axis: 'x',
        left: link.source.index,
        right: filteredNode.index,
        gap: serviceDistance + link.source.width / 2 + options.linkDistance,
        equality: true,
      });
```

Reply:
We need to show the event sink on the right side of pub-sub nodes, so we need these changes.